array-reduce/main.js

```diff
@@ -0,0 +1,37 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+const sum = numbers.reduce((accumulator, currentValue) => accumulator + currentValue);
+console.log('sum', sum);
+
+const product = numbers.reduce((accumulator, currentValue) => accumulator * currentValue);
+console.log('product', product);
+
+const balance = account.reduce((accumulator, currentValue) => {
+  if (currentValue.type === 'withdrawal') {
+    accumulator.amount = accumulator.amount - currentValue.amount;
+  } else {
+    accumulator.amount = accumulator.amount + currentValue.amount;
+  }
+  return accumulator;
+});
+console.log('balance', balance);
```

Comment:
Balance should be a number and not an object. You just overwrote the amount of the first transaction. Try an `initialValue` of `0`.